server/controllers/connectedAccounts.js

```diff
@@ -1,134 +1,109 @@
+
+      await userCollective.save();
+
+      let connectedAccount = await models.ConnectedAccount.findOne({
+        where: { service, CollectiveId: userCollective.id },
+      });
+      if (!connectedAccount) {
+        connectedAccount = await models.ConnectedAccount.create({
+          service,
+          CollectiveId: userCollective.id,
+          clientId: profile.id,
+          data: profile,
+          CreatedByUserId: req.remoteUser.id,
+        });
+      }
+      await connectedAccount.update({
+        username: data.profile.username,
+        token: accessToken,
+      });
+
+      const token = req.remoteUser.generateConnectedAccountVerifiedToken(connectedAccount.id, data.profile.username);
+      if (context === 'createCollective') {
+        res.redirect(`${config.host.website}/create/opensource?token=${token}`);
       } else {
-        fetchUserPromise = User.findOne({
-          where: {
-            email: { [Op.in]: emails.map(email => email.toLowerCase()) },
-          },
-        }).then(
-          u =>
-            u ||
-            User.createUserWithCollective({
-              name: profile.name || profile.login,
-              image,
-              email: emails[0],
-            }),
-        );
+        res.redirect(`${config.host.website}/${userCollective.slug}/edit/connected-accounts`);
       }
-      return fetchUserPromise
-        .then(u => {
-          user = u;
-          attrs.CollectiveId = user.CollectiveId;
-          attrs.clientId = profile.id;
-          attrs.data = profile;
-          attrs.CreatedByUserId = user.id;
-          return models.Collective.findByPk(user.CollectiveId);
-        })
-        .then(c => {
-          userCollective = c;
-          userCollective.description = userCollective.description || profile.bio;
-          userCollective.locationName = userCollective.locationName || profile.location;
-          userCollective.website = userCollective.website || profile.blog || profile.html_url;
-          userCollective.image = userCollective.image || image;
-          userCollective.githubHandle = data.profile.username;
-          userCollective.save();
-        })
-        .then(() =>
-          ConnectedAccount.findOne({
-            where: { service, CollectiveId: user.CollectiveId },
-          }),
-        )
-        .then(ca => ca || ConnectedAccount.create(attrs))
-        .then(ca => {
-          caId = ca.id;
-          return ca.update({
-            username: data.profile.username,
-            token: accessToken,
-          });
-        })
-        .then(() => {
-          const token = user.generateConnectedAccountVerifiedToken(caId, data.profile.username);
-          const newLocation = redirect
-            ? `${redirect}?token=${token}`
-            : `${config.host.website}/github/apply/${token}?utm_source=${utmSource}`;
-
-          res.redirect(newLocation);
-        })
-        .catch(next);
+
+      break;
     }
 
     case 'meetup':
```

Comment:
I think we can remove the `meetup` integration there, it's already removed from the frontend since the release of the new CCF

Reply:
Yep, will remove in a follow up PR.